src/bp/core/converse/converse-router.ts

```diff
@@ -89,6 +96,22 @@ export class ConverseRouter extends CustomRouter {
     )
   }
 
+  private async shouldRespondToPublicEndpoint(req, res, next) {
+    const { botId } = req.params
+
+    const enabledForBot = (await this.configProvider.getBotConfig(botId)).converse?.enableUnsecuredEndPoint ?? true
+    if (!enabledForBot) {
+      return res.status(401).send(`Unauthenticated Converse API is disabled for ${botId}`)
```

Comment:
I think 404 (without response body) would be a more appropriate response.

Reply:
I disagree on 404, it hinders the true situation and we end up investigating the wrong thing. Not sure 401 is the right choice, but maybe another one?

Reply:
I disagree on 404, it hinders the true error behind and we spend more time investigating. 401 may be not appropriate, but so is 404

Reply:
403 should do the trick